saleor/discount/utils.py

```diff
@@ -17,37 +20,28 @@ def decrease_voucher_usage(voucher):
     voucher.save(update_fields=["used"])
 
 
-def are_product_collections_on_sale(product, sale):
-    """Checks if any collection is on sale."""
-    discounted_collections = set(sale.collections.all())
-    product_collections = set(product.collections.all())
-    return set(product_collections).intersection(discounted_collections)
-
-
-def is_category_on_sale(category, sale):
-    """Check if category is descendant of one of categories on sale."""
-    discounted_categories = set(sale.categories.all())
-    return any(
-        [category.is_descendant_of(c, include_self=True) for c in discounted_categories]
-    )
+def are_product_collections_on_sale(product, discount: DiscountInfo):
+    """Check if any collection is on sale."""
+    discounted_collections = discount.collection_ids
+    product_collections = set(c.id for c in product.collections.all())
+    return product_collections.intersection(discounted_collections)
 
 
-def get_product_discount_on_sale(sale, product):
+def get_product_discount_on_sale(discount: DiscountInfo, product):
```

Comment:
For consistency we should place the `discount` argument as the second one.

Reply:
We should but this is an unrelated change.